validation/genesis/genesis-allocs_test.go

```diff
@@ -176,6 +177,18 @@ func testGenesisAllocs(t *testing.T, chain *ChainConfig) {
 	gotData, err := json.MarshalIndent(g.Alloc, "", " ")
 	require.NoError(t, err)
 
+	// special handling of weth9 for op-sepolia
+	// We've observed that the contract [metadata hash](https://docs.soliditylang.org/en/latest/metadata.html)
+	// does not match for the bytecode that is generated and what's in the superchain-registry.
+	// The issue is likely a difference in compiler settings when the contract artifacts were generated and
+	// stored in the superchain registry. To account for this, we trim the metadata hash portion of the
+	// weth9 contract bytecode before writing to file/comparing the outputs
+	if chainId == uint64(11155420) {
```

Comment:
I am wondering if to be extra safe here we require the commitID to be one that we are confident leads to bytecodes with metadata appended, since that is not guaranteed https://docs.soliditylang.org/en/latest/metadata.html. This would avoid issues if we ever change the commit we are using, we would need to whitelist the new commit for it to be treated in this way?

Reply:
Yeah, wouldn't hurt. I've added this too.